### PR TITLE
Add fontawesome kit to root.tsx

### PR DESCRIPTION
**Pull request by nikochiko**
This commit adds fontawesome kit JS to the root template.

It is added directly to the root.tsx page rather than as components because fontawesome needs the script to be put into the <head> tag, whereas otherwise, it's only the styles that go into `<head>` and the scripts go at the bottom of `<body>`.

### CODE REVIEW

File: app/root.tsx [hunk 1]
```diff
@@ -43,6 +43,7 @@ export default function App() {
         <meta name="viewport" content="width=device-width,initial-scale=1" />
         <Meta />
         <Links />
+        <script src="https://kit.fontawesome.com/8af9787bd5.js" crossorigin="anonymous"></script>
```
Comment (devxpy) [inline app/root.tsx:46]: do we really wanna load this before the body? would slow down the page no?

Comment (nikochiko) [reply app/root.tsx:46]: I thought so too, but it doesn't and is the recommended way. I think that has something to do with loading the fonts before the page renders, and that makes it faster instead.

Comment (devxpy) [reply app/root.tsx:46]: Put it here instead to follow remix convention? https://github.com/GooeyAI/gooey-ui/blob/2108ea1816330dfe0ded9ac2222cc39a541b2d1c/app/app.tsx#L39

Comment (nikochiko) [reply app/root.tsx:46]: Can we put <script> inside that?
I thought that is only for `<link>`s.

**Pull request merged**